lib/tasks/evm_application.rb

```diff
@@ -114,6 +114,149 @@ def self.output_workers_status(servers)
+        return
+      end
+
+      workers = servers.flat_map do |server|
+        # populate miq_server to avoid another server lookup. each will return the workers
+        server.miq_workers.each { |w| w.association(:miq_server).target = server }
+      end
+
+      show_overview_with_callouts(servers_overview(servers))
+
+      if my_server
+        puts "", "Local server: #{my_server.name || "UNNAMED"}, Region: #{my_server.region_number}"
+      end
+
+      puts "", ""
+      show_overview_with_callouts(workers_overview(workers))
+    end
+
+    def show_overview_with_callouts(data)
+      details = redundant_columns(data)
+      puts data.tableize(:columns => (data.first.keys - details.keys))
+      if details.present?
+        # dont give headsup for empty values
+        heads_up = details.select { |n, v| n == "Rgn" || (v != 0 && v.present?) }
+        puts "", "For all rows: #{heads_up.map { |n, v| "#{n}=#{v}" }.join(", ")}" if heads_up.present?
+      end
+    end
+
+    def compact_date(date)
+      return "" unless date
+      (date < 1.day.ago) ? date.strftime("%Y-%m-%d") : date.strftime("%H:%M:%S%Z")
+    end
+
+    STATUS_LOOKUP = {
+      "started"        => "Up",
+      "starting"       => "Up",
+      "aborted"        => "Down",
+      "stopped"        => "Down",
+      "not responding" => "?"
+    }
+
+    def compact_status(server)
+      STATUS_LOOKUP[server.status] || server.status || ""
+    end
+
+    # @param sw [MiqServer|MiqWorker] A server or worker object
+    def compact_memory(sw)
+      mb_usage = sw.try(:unique_set_size) || sw.try(:proportional_set_size) || sw.memory_usage
+      return "" unless mb_usage
+      mb_threshold = sw.try(:worker_settings).try("[]", :memory_threshold)
+      mb_threshold ? "#{mb_usage / 1.megabyte}/#{mb_threshold / 1.megabyte}" : (mb_usage / 1.megabyte).to_s
+    end
+
+    def compact_worker_type(name)
+      name && name.gsub(/(ManageIQ::Providers::|Manager|Worker|Miq)/, '')
+    end
+
+    def compact_queue(queue_name, uri)
+      if queue_name.nil?
+        # would like roles to come through
+        if uri
+          uri_parts = uri.split(":")
+          [uri_parts.first, uri_parts.last].join(":")
+        end
+      elsif queue_name.kind_of?(Array)
+        queue_name.join(":")
+      else
+        queue_name
+      end
+    end
+
+    def server_info(s, ws)
+      {
+        "Rgn"        => s.region_number,
+        "Zone"       => s.zone.name,
+        "Status"     => compact_status(s),
+        "Server"     => s.name || "", # s.hostname
+        "Mstr"       => s.is_master? ? "*" : "",
+        "Workers"    => ws ? ws.size : "",
+        "Version"    => s.version,
+#         "Build"      => s.build,
+        "Started On" => compact_date(s.started_on),
+        "Heartbeat"  => compact_date(s.last_heartbeat),
+        "Ram(mb)"    => compact_memory(s),
+        "Roles"      => (s.active_role_names.size == @role_count) ? "*" : s.active_role_names.join(':'),
+      }
```

Comment:
This feels very similar to the `output_servers_status` method above.  Can we make `rake evm:status` output what you want here and get rid of some of the duplication?  If we really want to keep both, can they both share the same base labels and values but only specify the different labels and values?